bundles/org.openhab.core.addon.marketplace/src/main/java/org/openhab/core/addon/marketplace/internal/community/CommunityMarketplaceAddonService.java

```diff
@@ -278,64 +280,69 @@ private String getContentType(@Nullable Integer category, List<String> tags) {
-        int likeCount = topic.likeCount;
-        int views = topic.views;
-        int postsCount = topic.postsCount;
-        Date createdDate = topic.createdAt;
-        String author = "";
-        for (DiscoursePosterInfo posterInfo : topic.posters) {
-            if (posterInfo.description.contains("Original Poster")) {
-                author = users.stream().filter(u -> u.id.equals(posterInfo.userId)).findFirst().get().name;
+            String link = COMMUNITY_TOPIC_URL + topic.id.toString();
+            int likeCount = topic.likeCount;
+            int views = topic.views;
+            int postsCount = topic.postsCount;
+            Date createdDate = topic.createdAt;
+            String author = "";
+            for (DiscoursePosterInfo posterInfo : topic.posters) {
+                if (posterInfo.description.contains("Original Poster")) {
+                    author = users.stream().filter(u -> u.id.equals(posterInfo.userId)).findFirst().get().name;
+                }
             }
-        }
 
-        String maturity = tags.stream().filter(CODE_MATURITY_LEVELS::contains).findAny().orElse(null);
+            String maturity = tags.stream().filter(CODE_MATURITY_LEVELS::contains).findAny().orElse(null);
 
-        Map<String, Object> properties = Map.of("created_at", createdDate, //
-                "like_count", likeCount, //
-                "views", views, //
-                "posts_count", postsCount, //
-                "tags", tags.toArray(String[]::new));
+            Map<String, Object> properties = Map.of("created_at", createdDate, //
+                    "like_count", likeCount, //
+                    "views", views, //
+                    "posts_count", postsCount, //
+                    "tags", tags.toArray(String[]::new));
 
-        // try to use a handler to determine if the add-on is installed
-        boolean installed = addonHandlers.stream()
-                .anyMatch(handler -> handler.supports(type, contentType) && handler.isInstalled(uid));
+            // try to use a handler to determine if the add-on is installed
+            boolean installed = addonHandlers.stream()
+                    .anyMatch(handler -> handler.supports(type, contentType) && handler.isInstalled(uid));
 
-        return Addon.create(uid).withType(type).withId(id).withContentType(contentType).withImageLink(topic.imageUrl)
-                .withAuthor(author).withProperties(properties).withLabel(title).withInstalled(installed)
-                .withMaturity(maturity).withCompatible(compatible).withLink(link).build();
+            return Addon.create(uid).withType(type).withId(id).withContentType(contentType)
+                    .withImageLink(topic.imageUrl).withAuthor(author).withProperties(properties).withLabel(title)
+                    .withInstalled(installed).withMaturity(maturity).withCompatible(compatible).withLink(link).build();
+        } catch (Exception e) {
```

Comment:
Don't catch `Exception`, better use `IllegalArgumentException` and log at DEBUG. If you want to make sure that this never fails, add a second `catch` clause and catch `RuntimeException` and log that at WARN.

Reply:
updated, hope I got it right ...